swift/app.py

```diff
@@ -84,3 +95,86 @@ def _receivedMessage(self, channelName: str, msg: str):
+              self.swiftcallRequested signal.
+            msg: The received swift-call result message.
+        """
+        try:
+            result = swift.loads(swift.SwiftcallResult, msg)
+        except json.JSONDecodeError as e:
+            print(f"{self}._receivedResult: {e}")
+        else:
+            self.swiftcall.update_result(request, result)
+
+
+class SwiftcallProxy:  # pylint: disable=too-few-public-methods
+    """A proxy for requesting swift-calls conveniently.
+    
+    Every attribute access is proxied, and if you try to call a method of this
+    object, it will emit a swift-call requesting signal instead.
+    If you get an attribute of this object, you will get a callable object which
+    does the same thing as calling a method of this object.
+    """
+
+    def __init__(self, requested: QObject):
+        """
+        Args:
+            requested: A pyqtSignal(str) which will be emitted when a proxied
+              method call is invoked. See BaseApp.swiftcallRequested.
+        """
+        self.requested = requested
+        self.results: Dict[str, swift.SwiftcallResult] = {}
+
+    def __getattr__(self, call: str) -> Callable:
+        """Returns a callable object which emits a swift-call requesting signal.
+
+        Args:
+            call: The name of the swift-call.
+        """
+        def proxy(**args: Any) -> swift.SwiftcallResult:
+            """Emits a swift-call request signal with the given arguments.
+
+            It saves the returned result to self.results dictionary, so when
+            self.returned signal is emitted, i.e., the swift-call result is received,
+            it will update the result object contents.
+
+            Args:
+                **args: The arguments for the swift-call, all as keyword arguments.
+                  If an argument is a swift.Serializable instance, it will be
+                  converted to a JSON string by swift.dumps().
+
+            Returns:
+                A swift-call result object to keep tracking the result.
+            """
+            for name, arg in args.items():
+                if isinstance(arg, swift.Serializable):
+                    args[name] = swift.dumps(arg)
+            info = swift.SwiftcallInfo(call=call, args=args)
+            result = swift.SwiftcallResult(done=False, success=False)
+            msg = swift.dumps(info)
+            self.results[msg] = result
+            self.requested.emit(msg)
+            return result
+        return proxy
+
+    def update_result(self, request: str, result: swift.SwiftcallResult):
+        """Updates the result for the request parsing the received message.
+
+        Args:
+            request: The request message that has been sent to Swift.
+            result: The received result object.
+        """
+        _result = self.results.get(request, None)
+        if _result is None:
+            print(f"SwiftcallProxy._update_result: Failed to find a result for {request}.")
```

Comment:
It is just a coding style, but how about modifying `SwiftcallProxy._update_result()`.
So far, we have adopted this form.

Reply:
Oh, I didn't know that. Thanks!
Moreover I will change that `.update_result()`, as the method is public.